common/gps.h

```diff
@@ -0,0 +1,47 @@
+//GPS
+
+
+#include "nmea_parser.h"
+#include "serial_reader.h"
+#include "../third_party/units.h"
+#include "map_coordinate.h"
+
+namespace BoBRobotics 
+{
+namespace GPS 
+{
+
+class Gps {
+
+    private:
+    
+    SerialReader sreader;
+    const char *m_device_path;
+    
+    public:
+
+    Gps(const char *device_path) : m_device_path(device_path) {
+
+        try {
+            sreader.connect(device_path);
+            if (!sreader.isConnected()) throw "device not found";
+            m_device_path = device_path;
+        } catch(const char* error) {
```

Comment:
I'd throw a ``std::runtime_error`` (``#include <stdexcept>``) rather than a ``const char *``, even though it is technically valid c++

Reply:
Also I wouldn't actually catch and rethrow the exception here -- just let it pass up the call stack.

Is the issue that if you don't catch it here, you don't get to see what the error is? I know this is an issue on Windows, but not sure about Macs.